addons/point_of_sale/models/pos_order.py

```diff
@@ -662,12 +662,13 @@ def action_pos_order_cancel(self):
     def _apply_invoice_payments(self):
         receivable_account = self.env["res.partner"]._find_accounting_partner(self.partner_id).property_account_receivable_id
         payment_moves = self.payment_ids.sudo().with_company(self.company_id)._create_payment_moves()
-        invoice_receivable = self.account_move.line_ids.filtered(lambda line: line.account_id == receivable_account)
+        invoice_receivables = self.account_move.line_ids.filtered(lambda line: line.account_id == receivable_account)
```

Comment:
can't we do something like this?
```suggestion
        if receivable_account.reconcile
            invoice_receivables = self.account_move.line_ids.filtered(lambda line: line.account_id == receivable_account and not line.reconciled)
            if invoice receivables:
                payment_receivables = payment_moves.mapped('line_ids').filtered(lambda line: line.account_id == receivable_account and line.partner_id)
                (invoice_receivable | payment_receivables).sudo().with_company(self.company_id).reconcile()
```